### PR TITLE
Add initial React Labs June 2022 blog post

### PR DESCRIPTION
**Pull request by poteto**
# Overview
Adds a React Labs blog post sharing insights into what the React team has been working on.

Todos:

- [x] @acdlite tweaks to Offscreen section

### CODE REVIEW

**Review by rickhanlonii — APPROVED** [review]
🚀

File: content/blog/2022-06-15-react-labs-what-we-have-been-working-on-june-2022.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+* **Instant transitions.** Some routing frameworks already prefetch data to speed up subsequent navigations, like when hovering over a link. With Offscreen, they'll also be able to prerender the next screen in the background.
+* **Reusable state.** Similarly, when navigating between routes or tabs, you can use Offscreen to preserve the state of the previous screen so you can switch back and pick up where you left off.
+* **Virtualized list rendering.** When displaying large lists of items, virtualized list frameworks will prerender more rows than are currently visible. You can use Offscreen to prerender the hidden rows at a lower priority than the visible items in the list.
+
+## Transition Tracing {#transition-tracing}
+
+Currently, React has two profiling tools. The old Profiler shows an overview of all the commits in a profiling session. For each commit, it also shows all components that rendered and the amount of time it took for them to render. The new Timeline Profiler shows when components schedule updates and when React works on these updates. Both of these profilers help developers identify performance problems in their code.
```
[the remark this answers is not in the document]
Comment (poteto) [reply content/blog/2022-06-15-react-labs-what-we-have-been-working-on-june-2022.md:55]: @lunaruan

[the remark this answers is not in the document]
Comment (lunaruan) [reply content/blog/2022-06-15-react-labs-what-we-have-been-working-on-june-2022.md:55]: I think transition tracing will probably be key here in making timeline profiler more understandable

[the remark this answers is not in the document]
Comment (gaearon) [reply content/blog/2022-06-15-react-labs-what-we-have-been-working-on-june-2022.md]: hmm i feel like mentioning it without linking is more confusing (even if we don't want to straight-up tell people to use it) but your call

[the remark this answers is not in the document]
Comment (lunaruan) [reply content/blog/2022-06-15-react-labs-what-we-have-been-working-on-june-2022.md]: I linked it but called it "beta" version so people don't complain lol

File: content/blog/2022-06-15-react-labs-what-we-have-been-working-on-june-2022.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+Offscreen is a low level capability that unlocks high level features. Similar to React's other concurrent features like `startTransition`, in most cases you won't interact with the Offscreen API directly, but instead via an opinionated framework to implement patterns like:
+
+* **Instant transitions.** Some routing frameworks already prefetch data to speed up subsequent navigations, like when hovering over a link. With Offscreen, they'll also be able to prerender the next screen in the background.
+* **Reusable state.** Similarly, when navigating between routes or tabs, you can use Offscreen to preserve the state of the previous screen so you can switch back and pick up where you left off.
+* **Virtualized list rendering.** When displaying large lists of items, virtualized list frameworks will prerender more rows than are currently visible. You can use Offscreen to prerender the hidden rows at a lower priority than the visible items in the list.
```
Comment (acdlite) [inline content/blog/2022-06-15-react-labs-what-we-have-been-working-on-june-2022.md:51]:
```suggestion
* **Virtualized list rendering.** When displaying large lists of items, virtualized list frameworks will prerender more rows than are currently visible. You can use Offscreen to prerender the hidden rows at a lower priority than the visible items in the list.
```
```suggestion
* **Virtualized list rendering.** When displaying large lists of items, virtualized list frameworks will prerender more rows than are currently visible. You can use Offscreen to prerender the hidden rows at a lower priority than the visible items in the list.
**Backgrounded content.** We're also exploring a related feature for deprioritizing content in the background without hiding it, like when displaying a modal overlay.
```

File: content/blog/2022-06-15-react-labs-what-we-have-been-working-on-june-2022.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+To set expectations, this is not a roadmap with clear timelines. Many of these projects are under active research and are difficult to put concrete ship dates on. They may possibly never even ship in their current iteration depending on what we learn. Instead, we want to share with you the problem spaces we’re actively thinking about, and what we’ve learned so far.
+
+## Server Components {#server-components}
+
+We announced an [experimental demo of React Server Components](https://reactjs.org/blog/2020/12/21/data-fetching-with-react-server-components.html) (RSC) in December 2020. Since then we’ve been finishing up its dependencies in React 18, and working on changes inspired by experimental feedback.
+
+In particular, we’re abandoning the idea of having forked I/O libraries (eg react-fetch), and instead adopting an async/await model for better compatibility. This doesn’t technically block RSC’s release because you can also use routers for data fetching. Another change is that we’re also moving away from the file extension approach in favor of [annotating boundaries](https://github.com/reactjs/rfcs/pull/189#issuecomment-1116482278).
```
Comment (PupoSDC) [inline content/blog/2022-06-15-react-labs-what-we-have-been-working-on-june-2022.md:16]: Is there any discussion available for dropping `react-fetch` in favour of an async/await model, and how that model would look like? I would like to learn more about it 😄

**Pull request merged**